e2e/test/scenarios/native/native-database-source.cy.spec.js

```diff
@@ -0,0 +1,247 @@
+import { SAMPLE_DB_ID } from "e2e/support/cypress_data";
+import {
+  restore,
+  popover,
+  addPostgresDatabase,
+  POPOVER_ELEMENT,
+} from "e2e/support/helpers";
+
+const PG_DB_ID = 2;
+const mongoName = "QA Mongo4";
+const additionalPG = "New Database";
+
+describe(
+  "scenarios > question > native > database source",
+  { tags: "@external" },
+  () => {
+    beforeEach(() => {
+      cy.intercept("PUT", "/api/setting/last-used-database-id").as(
+        "persistDatabase",
+      );
+
+      restore("postgres-12");
+      cy.signInAsAdmin();
+    });
+
+    it("smoketest: persisting last used database should work, and it should be user-specific setting", () => {
+      const adminPersistedDatabase = "QA Postgres12";
+      const userPersistedDatabase = "Sample Database";
+
+      startNativeQuestion();
+      assertNoDatabaseSelected();
+
+      selectDatabase(adminPersistedDatabase);
+
+      startNativeQuestion();
+      assertSelectedDatabase(adminPersistedDatabase);
+
+      cy.signOut();
+      cy.signInAsNormalUser();
+
+      startNativeQuestion();
+      assertNoDatabaseSelected();
+
+      selectDatabase(userPersistedDatabase);
+
+      startNativeQuestion();
+      assertSelectedDatabase(userPersistedDatabase);
+
+      cy.signOut();
+      cy.signInAsAdmin();
+
+      startNativeQuestion();
+      assertSelectedDatabase(adminPersistedDatabase);
+    });
+
+    it("deleting previously persisted database should result in the new database selection prompt", () => {
+      addPostgresDatabase(additionalPG);
+
+      startNativeQuestion();
+      assertNoDatabaseSelected();
+
+      selectDatabase(additionalPG);
+
+      cy.log("Delete previously persisted database.");
+      cy.get("@postgresID").then(databaseId => {
+        cy.request("DELETE", `/api/database/${databaseId}`);
+      });
+
+      startNativeQuestion();
+      assertNoDatabaseSelected();
```

Comment:
I wonder if this test is going to pass if `startNativeQuestion` does not call `cy.visit`.
It should not, and we should ensure that the setting value is not stale (i.e. is refetched) when visiting native query editor.

Reply:
+1

Reply:
Unfortunately, our caching logic creates all sorts of issues not just here, but in a lot of other places. Unless we do a hard refresh (which visiting a home page achieves in Cypress), most of this will not work.

This also applies to [the other comment](https://github.com/metabase/metabase/pull/38986#discussion_r1503801570) and [my answer](https://github.com/metabase/metabase/pull/38986#discussion_r1503871038) there.